sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpEvent.kt

```diff
@@ -0,0 +1,142 @@
+package io.sentry.android.okhttp
+
+import io.sentry.Breadcrumb
+import io.sentry.Hint
+import io.sentry.IHub
+import io.sentry.ISpan
+import io.sentry.SpanStatus
+import io.sentry.TypeCheckHint
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.CONNECTION_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.CONNECT_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.REQUEST_BODY_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.REQUEST_HEADERS_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.RESPONSE_BODY_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.RESPONSE_HEADERS_EVENT
+import io.sentry.android.okhttp.SentryOkHttpEventListener.Companion.SECURE_CONNECT_EVENT
+import io.sentry.util.UrlUtils
+import okhttp3.Request
+import okhttp3.Response
+
+internal class SentryOkHttpEvent(private val hub: IHub, private val request: Request) {
+    private val eventSpans: MutableMap<String, ISpan> = HashMap()
```

Comment:
M: Should this be thread-safe? can it be called from multiple threads?

Reply:
This is called by the event listener only, and I think each "event" call is done in a specific thread per `Call`. However, using a `ConcurrentHashMap` makes no harm here